kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/internal/KubernetesDeserializer.java

```diff
@@ -248,16 +248,20 @@ private String getClassName(String key) {
             // If only one class found, return it
             if (possibleResults.size() == 1) {
                 return possibleResults.get(0);
-            }
-
-            // Compare with apiVersions being compared for set of classes found
-            for (Class<? extends KubernetesResource> result : possibleResults) {
+            } else if (possibleResults.size() > 1) {
+              // Compare with apiVersions being compared for set of classes found
+              for (Class<? extends KubernetesResource> result : possibleResults) {
                 String defaultKeyFromClass = getKeyFromClass(result);
                 if (key.equals(defaultKeyFromClass)) {
-                    return result;
+                  return result;
                 }
+              }
+              // Not able to compare with apiVersions, perhaps it was not provided;
+              // return first item found
+              return possibleResults.get(0);
```

Comment:
Are we sure we want to return the first item found.
This means that `PACKAGES` must be sorted by order of preference (for packages providing same Types/Kinds). If so maybe we should document this somewhere.

Reply:
I am also not sure whether returning first item found is good or not. But I thought if user wanted some specific apiVersion related resource. He should be providing apiVersion inside object

Reply:
What I mean is that probably the PACKAGE array definition should have an n.b comment e.g.:
`// n.b. Packages sorted in order of precedence, deserialization of resources with no specific version will default to first available Class in one of these packages:`

Reply:
Also unit tests for conflicting classes would help enforce list of entries doesn't get messed up (and can probably avoid the comment)